### PR TITLE
Add docs for error code

### PR DESCRIPTION
**Pull request by kthui**
Related PRs:
https://github.com/triton-inference-server/python_backend/pull/292
https://github.com/triton-inference-server/server/pull/6228

Add docs on how to return an error code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -485,6 +485,21 @@ class TritonPythonModel:
...
 ```
 
+Starting from 23.09, `pb_utils.TritonError` may be constructed with an optional
+Triton error code on the second parameter. For example:
+
+```
+pb_utils.TritonError("The file is not found", pb_utils.TritonError.NOT_FOUND)
+```
+
+Supported error codes:
```
Comment (nv-kmcgill53) [inline README.md:495]: Does there need to be more description with these error codes or is it just a 1-to-1 mapping from the triton internals?

Comment (rmccorm4) [inline README.md:495]:
```suggestion
If no code is specified, `pb_utils.TritonError.INTERNAL` will be used by default.

Supported error codes:
```

**Review by Tabrizian — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -485,6 +485,23 @@ class TritonPythonModel:
         return responses
 ```
 
+Starting from 23.09, `pb_utils.TritonError` may be constructed with an optional
+Triton error code on the second parameter. For example:
+
+```
```
Comment (rmccorm4) [inline README.md:491]:
```suggestion
```python
```

**Review by rmccorm4 — APPROVED** [review]
Nice :100:

**Review by nv-kmcgill53 — APPROVED** [review]

**Pull request merged**